### PR TITLE
Geometry_Engine: Change variable name to minimumSegmentLength

### LINKED ISSUE #1173: Geometry_Engine: CleanPolyline variable name
#### Description:

 I recommend we replace name from distanceTolerance to minimumSegmentLength as below ,I will keep right now as we need to fixed also in  `XML Toolkit`. We discussed with @pawelbaran and @KonradStolarski  this is not tolerance but minimumSegmentLength so it is confusing for users. In case we have some point along line in the close distance we need to remove them for Environment it is 0.02. 

` static Polyline CleanPolyline(this Polyline polyline, double angleTolerance = Tolerance.Angle, double minimumSegmentLength = 0.02)`

#### How to replicate:

clarification of documentation and naming

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
### Issues addressed by this PR
Fixes #1173 

### Test files
[Existing](https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/01_Test%20Scripts/BHoM_Engine/Geometry_Engine/CleanPolyline.gh?csf=1&e=mpq5Kw)

### Changelog
 - Rename `distanceTolerance` to `minmumSegmentLength` in `CleanPolyline`
 - Add `RemoveLeastSignificantVertices` method separately from `CleanPolyline` for greater user control
 - Add `RemoveShortSegments` method separately from `CleanPolyline` for greater user control
 - Modify `CleanPolyline` to use the two separate methods
 - Add `tolerance` to all `centroid` methods

### Additional comments
Tolerance added to all Centroid methods per PR comment https://github.com/BHoM/BHoM_Engine/pull/1170#issuecomment-529340030

### CODE REVIEW

**Comment by FraserGreenroyd** [comment]
@pawelbaran see link above :smile:

File: Geometry_Engine/Query/Centroid.cs [hunk 1]
```diff
@@ -47,6 +47,40 @@ public static Point Centroid(this PolyCurve curve)
             return curve.Centroid(Tolerance.Distance);
         }
 
+        [Deprecated("2.4", "Deprecated to expose tolerance as optional parameter for greater control", null, "Centroid(this Ellipse ellipse, double tolerance = Tolerance.Distance)")]
+        public static Point Centroid(this Ellipse ellipse)
+        {
+            return ellipse.Centre;
```
Comment (pawelbaran) [inline Geometry_Engine/Query/Centroid.cs]: Could it be `return ellipse.Centroid(Tolerance.Distance);`?

File: Geometry_Engine/Query/Centroid.cs [hunk 2]
```diff
@@ -47,6 +47,40 @@ public static Point Centroid(this PolyCurve curve)
...
+        }
+
+        [Deprecated("2.4", "Deprecated to expose tolerance as optional parameter for greater control", null, "Centroid(this Circle circle, double tolerance = Tolerance.Distance)")]
+        public static Point Centroid(this Circle circle)
+        {
+            return circle.Centre;
+        }
```
Comment (pawelbaran) [inline Geometry_Engine/Query/Centroid.cs]: As above

File: Geometry_Engine/Query/Centroid.cs [hunk 3]
```diff
@@ -47,6 +47,40 @@ public static Point Centroid(this PolyCurve curve)
...
+
+        [Deprecated("2.4", "Deprecated to expose tolerance as optional parameter for greater control", null, "Centroid(this Line line, double tolerance = Tolerance.Distance)")]
+        public static Point Centroid(this Line line)
+        {
+            return line.PointAtParameter(0.5);
```
Comment (pawelbaran) [inline Geometry_Engine/Query/Centroid.cs]: As above

File: Geometry_Engine/Query/Centroid.cs [hunk 4]
```diff
@@ -47,6 +47,40 @@ public static Point Centroid(this PolyCurve curve)
...
+        }
+
+        [NotImplemented]
+        [Deprecated("2.4", "Deprecated to expose tolerance as optional parameter for greater control", null, "Centroid(this Arc arc, double tolerance = Tolerance.Distance)")]
+        public static Point Centroid(this Arc arc)
+        {
+            throw new NotImplementedException();
```
Comment (pawelbaran) [inline Geometry_Engine/Query/Centroid.cs]: As above

File: Geometry_Engine/Query/Centroid.cs [hunk 5]
```diff
@@ -47,6 +47,40 @@ public static Point Centroid(this PolyCurve curve)
...
+        }
+
+        [NotImplemented]
+        [Deprecated("2.4", "Deprecated to expose tolerance as optional parameter for greater control", null, "Centroid(this NurbsCurve nurbsCurve, double tolerance = Tolerance.Distance)")]
+        public static Point Centroid(this NurbsCurve nurbsCurve)
+        {
+            throw new NotImplementedException();
```
Comment (pawelbaran) [inline Geometry_Engine/Query/Centroid.cs]: As above

**Comment by FraserGreenroyd** [comment]
@pawelbaran done - see new changes and I have got someone on the case about the test script (I'm not sure where the original one went cause the original one was better!) :smile:

**Comment by FraserGreenroyd** [comment]
Perfect, thanks @pawelbaran - am happy as the changes you made were clean up to standards as discussed offline, so will get this merged :smile:

**Pull request merged**